### DISCUSSION TITLE
<ActionForm/> ----  multiple inputs,   automatically moving focus,   and set input to " " after submit

### DISCUSSION THREAD

**Discussion by yingweixiang** [leptos-rs/leptos#3137] [General]
I have several input areas on the <ActionForm>.  it's used to insert records into a database. 
questions:  
1)    how to let the cursor/focus jump automatically into next input area ?   (like, using a scanner to scan 3 barcodes into 3 input areas).
2)    how to reset the input areas to "" ? 

I'm new to leptos, thanks for your help.

**Comment by gbj** [comment]
With respect to its children, it is simply a wrapper around a regular `<form>` -- you can do this the same way you would with any JavaScript frontend framework.

That is to say: I don't know specifically how you'd implement it, but I assume some combination of
- `on:change` or `on:input` to detect when you've set each input to its value
- a `NodeRef` (or an element with an `id`) to get the next element, and then [`.focus()`](https://docs.rs/web-sys/latest/web_sys/struct.HtmlElement.html#method.focus) to focus it
- you could use `prop:value` to set the values to `""` reactively, or [`set_value`](https://docs.rs/web-sys/latest/web_sys/struct.HtmlInputElement.html#method.set_value) to do it imperatively

These chapters of the docs will be useful to read if you have not read them:
- [Forms and Inputs](https://book.leptos.dev/view/05_forms.html)
- The [`web-sys`](https://book.leptos.dev/web_sys.html) chapter

**Reply by yingweixiang** [reply]
Thanks for your kind help. 
I'll read those chapters in detail.
